packages/nx/src/config/project-graph.ts

```diff
@@ -8,25 +8,60 @@ import { NxJsonConfiguration } from './nx-json';
 /**
  * Some metadata about a file
  */
-export interface FileData {
+export interface BaseFileData {
   file: string;
   hash: string;
-  deps?: (string | [string, string])[];
 }
 
-export function fileDataDepTarget(dep: string | [string, string]) {
-  return typeof dep === 'string' ? dep : dep[0];
+/**
+ * Represents metadata about a file that belongs to a project.
+ */
+export interface ProjectFileData extends BaseFileData {
+  /**
+   * An array of dependencies. If an element is just a string,
+   * the dependency is assumed to be a static dependency targetting
+   * that string. If the element is a tuple, the first element inside of it
+   * is the target project, with the second element being the type of dependency.
+   */
+  deps?: (string | [target: string, type: string])[];
+}
+
+/**
+ * Represents metadata about a file that **does not** belong to a project.
+ */
+export interface NonProjectFileData extends BaseFileData {
```

Comment:
Make this more similar to the projectFileMap deps.

Add a tuple of `[src, target, type]`